src/main/java/org/yomigae/patterns/PositionedPattern.java

```diff
@@ -0,0 +1,44 @@
+package org.yomigae.patterns;
+
+import heronarts.lx.LX;
+import heronarts.lx.LXPattern;
+import heronarts.lx.model.LXPoint;
+import heronarts.lx.parameter.CompoundParameter;
+import heronarts.lx.parameter.EnumParameter;
+
+public abstract class PositionedPattern extends LXPattern {
+
+  public enum Axis {
+    X, Y
+  };
+
+  public final EnumParameter<Axis> axis = new EnumParameter<Axis>("Axis", Axis.X)
+    .setDescription("Which axis the pattern is drawn across");
+
+  // runs from -1 to 2 to allow the pattern to run fully off the model, even at full width
```

Comment:
Can you think of a better way to model this?  It would be nice if we could dynamically calculate the min and max of this pattern based on the `width` value, such that the extremes of the knob always mapped to exactly off-screen.